### PR TITLE
fix(igxHierarchicalGrid): Fixing navigation when parent has summary b…

### LINKED ISSUE #4416: In igxHierarchicalGrid it is not possible to navigate through the parent summaries by using "Tab", if the last child grid is expanded
## Description  
When the last child grid has been expanded, trying to use "Tab" key to go through the parent summaries does not work.

 * igniteui-angular version: 7.2.x
 * browser: N/A

## Steps to reproduce  

1. Run the igniteui-angular-samples repo with the lastest 7.2.x branch
2. Open http://localhost:4200/samples/hierarchical-grid/hierarchical-grid-summary
3. Expand the last row - Peter Taylor
4. Scroll to the bottom and select the first cell of the last row - Sleeping Dogs
5. Press "Tab" key 4 times

## Result  
Pressing Tab 3 times selects the last cell of the row, but pressing it 4th time does not navigate through the parent grid summaries. Tab key goes through the summaries only if the last row of the parent grid is not currently expanded.

## Expected result  
"Tab" key should navigate through the parent summaries irrespective of the expand/collapse  state of the last row.

### PR DESCRIPTION
**Pull request by MayaKirova**
…ut child does not.

Closes #4416   

### Additional information (check all that apply):
 - [x] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code
 - [ ] This PR includes API docs for newly added methods/properties
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

**Comment by mpavlinov** [comment]
@MayaKirova I found 2 problems:

#### Cannot read property 'childGridsData' of undefined
Steps

1. Expand "Ahmad Nazeri"
2. Focus "Kimmy McIlmorie"
3. Press Shift + Tab
4. Error is thrown

```
ERROR TypeError: Cannot read property 'childGridsData' of undefined
    at IgxHierarchicalGridComponent.push../projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.component.ts.IgxHierarchicalGridComponent.isChildGridRecord (hierarchical-grid.component.ts:535)
    at IgxHierarchicalGridNavigationService.push../projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid-navigation.service.ts.IgxHierarchicalGridNavigationService.getIsChildAtIndex (hierarchical-grid-navigation.service.ts:54)
    at IgxHierarchicalGridNavigationService.push../projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid-navigation.service.ts.IgxHierarchicalGridNavigationService.goToLastCell (hierarchical-grid-navigation.service.ts:193)
    at IgxHierarchicalGridNavigationService.push../projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid-navigation.service.ts.IgxHierarchicalGridNavigationService.performShiftTabIntoChild (hierarchical-grid-navigation.service.ts:453)
    at IgxHierarchicalGridNavigationService.push../projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid-navigation.service.ts.IgxHierarchicalGridNavigationService.performShiftTabKey (hierarchical-grid-navigation.service.ts:396)
```

#### Cannot read property 'length' of null
Steps
1. Expand "Alicia Stanger"
2. Focus "Peter Taylor"
3. Press Shift + Tab

IgxHierarchicalRowComponent.html:34 ERROR TypeError: Cannot read property 'length' of null
    at IgxHierarchicalGridNavigationService.push../projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid-navigation.service.ts.IgxHierarchicalGridNavigationService.goToLastCell (hierarchical-grid-navigation.service.ts:192)
    at IgxHierarchicalGridNavigationService.push../projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid-navigation.service.ts.IgxHierarchicalGridNavigationService.performShiftTabIntoChild (hierarchical-grid-navigation.service.ts:453)
    at IgxHierarchicalGridNavigationService.push../projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid-navigation.service.ts.IgxHierarchicalGridNavigationService.performShiftTabKey (hierarchical-grid-navigation.service.ts:396)
    at IgxHierarchicalGridCellComponent.push../projects/igniteui-angular/src/lib/grids/cell.component.ts.IgxGridCellComponent.handleTab (cell.component.ts:768)
    at IgxHierarchicalGridCellComponent.push../projects/igniteui-angular/src/lib/grids/cell.component.ts.IgxGridCellComponent.dispatchEvent (cell.component.ts:845)
    at IgxHierarchicalGridCellComponent.push../projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-cell.component.ts.IgxHierarchicalGridCellComponent.dispatchEvent (hierarchical-cell.component.ts:111)

File: projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid-navigation.service.ts [hunk 1]
```diff
@@ -284,27 +284,13 @@ export class IgxHierarchicalGridNavigationService extends IgxGridNavigationServi
                 } else {
                     this.focusNextChildDOMElem(currentRowEl, this.grid);
                 }
+            } else if (this.grid.parent && this.grid.parent.summariesRowList.toArray().length > 0) {
```
Comment (ChronosSF) [inline projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid-navigation.service.ts]: `summariesRowList` should be a QueryList which also has a length property. Don't think you need to convert to Array

File: projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid-navigation.service.ts [hunk 2]
```diff
@@ -313,6 +299,26 @@ export class IgxHierarchicalGridNavigationService extends IgxGridNavigationServi
         }
     }
 
+    private _navigateToNextParentRow(currentRowEl) {
+         // next is parent summary or next parent row
+         const parent = this.grid.parent;
+         const parentHasSummary = parent.summariesRowList.toArray().length > 0;
```
Comment (ChronosSF) [inline projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid-navigation.service.ts]: Same as the above

File: projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid-navigation.service.ts [hunk 3]
```diff
@@ -313,6 +299,26 @@ export class IgxHierarchicalGridNavigationService extends IgxGridNavigationServi
...
+         const parentRowIndex = parseInt(
+             this.getClosestElemByTag(currentRowEl, 'igx-child-grid-row').parentNode.getAttribute('data-rowindex'), 10);
+         const isLastRowInParent = parent.verticalScrollContainer.igxForOf.length - 1 === parentRowIndex;
+         // check if next is sibling
+         const childRowContainer = this.getChildGridRowContainer(this.grid);
+         const nextIsSiblingChild = !!childRowContainer.nextElementSibling;
+         if (isLastRowInParent && parentHasSummary && !nextIsSiblingChild) {
+             // next is parent summary
+             const parentSummary = parent.summariesRowList.toArray()[0].nativeElement;
+             parent.navigation.focusNextRow(parentSummary, 0, this.grid.rootGrid, true);
```
Comment (ChronosSF) [inline projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid-navigation.service.ts]: you sure the third param should be this.grid.rootGrid? Won't this fail if parent and rootGrid are different?

File: projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid-navigation.service.ts [hunk 4]
```diff
@@ -437,12 +443,29 @@ export class IgxHierarchicalGridNavigationService extends IgxGridNavigationServi
              childGrid.allowFiltering && childGrid.filterMode === FilterMode.quickFilter) {
                  // move to filter cell
                 childGrid.navigation.moveFocusToFilterCell();
+            } else if (childGrid.rowList.toArray().length === 0) {
```
Comment (ChronosSF) [inline projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid-navigation.service.ts]: Again QueryLists have length

File: projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid-navigation.service.ts [hunk 5]
```diff
@@ -694,7 +717,12 @@ export class IgxHierarchicalGridNavigationService extends IgxGridNavigationServi
         grid.parentVirtDir.onChunkLoad
             .pipe(first())
             .subscribe(callBackFunc);
-        grid.dataRowList.toArray()[0].virtDirRow.scrollTo(unpinnedIndex);
+        if (grid.dataRowList.toArray().length > 0) {
```
Comment (ChronosSF) [inline projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid-navigation.service.ts]: QueryLists have length

File: projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid-navigation.service.ts [hunk 6]
```diff
@@ -694,7 +717,12 @@ export class IgxHierarchicalGridNavigationService extends IgxGridNavigationServi
...
+            grid.dataRowList.toArray()[0].virtDirRow.scrollTo(unpinnedIndex);
```
Comment (ChronosSF) [inline projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid-navigation.service.ts]: QueryLists have a `first` property

File: projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid-navigation.service.ts [hunk 7]
```diff
@@ -313,6 +299,26 @@ export class IgxHierarchicalGridNavigationService extends IgxGridNavigationServi
         }
     }
 
+    private _navigateToNextParentRow(currentRowEl) {
```
Comment (ChronosSF) [inline projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid-navigation.service.ts]: Please add param and return type

**Pull request merged**